tests/src/main/java/io/kestra/core/runners/TestRunner.java

```diff
@@ -25,6 +29,8 @@
 @Slf4j
 @Singleton
 public class TestRunner implements Runnable, AutoCloseable {
+    public static final ThreadLocal<List<Execution>> testExecutions = ThreadLocal.withInitial(ArrayList::new);
```

Comment:
Why don't you move this thread local inside the TestQueueFactory?
This would avoid setting it so you would directly access it via the factory